test/e2e/azure_test.go

```diff
@@ -937,6 +955,28 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 
+	Context("Creating an AKS cluster with the ASO API [Managed Kubernetes][OPTIONAL]", func() {
```

Comment:
I'd sort of prefer we remove the "OPTIONAL" label. Do we think there's a risk of adding PR friction? I would expect the new API to be operationally equivalent to the existing AKS CAPZ API.

Reply:
I don't suspect this test will be flakier than any of the other tests. I was just generally thinking that experimental things probably don't need to be required. I'll change it though and we can reevaluate if this becomes an issue.